### ISSUE TITLE
coredns svc connection timed out; no servers could be reached

### ISSUE THREAD

**Issue by Mihai-CMM** [kubernetes/dns#550]
Hello saw this a few times in the past . I wonder if anyone can help with it on my particular case please
I have a kubespray deployed cluster and i tried with cilium, calico  and flannel always with the same result.
I can get DNS A records if I  query a coredns pod but not if i query the service itself.

`kubectl exec -i -t dnsutils -- nslookup gsp.ro

;; connection timed out; no servers could be reached
`

` kubectl get pod -o wide -n kube-system| grep core
coredns-69dfc8446-b2qkx                    1/1     Running   0          3h35m   192.168.88.1    dl-mstr2-cr-sit   <none>           <none>
coredns-69dfc8446-nkl9l                    1/1     Running   0          3h35m   192.168.100.1   dl-mstr1-cr-sit   <none>           <none>

kubectl exec -i -t dnsutils -- nslookup gsp.ro 192.168.88.1
Server:         192.168.88.1
Address:        192.168.88.1#53

Non-authoritative answer:
Name:   gsp.ro
Address: 172.67.73.218
Name:   gsp.ro
Address: 104.26.5.123
Name:   gsp.ro
Address: 104.26.4.123

`
It seems only the svc does not properly work 

`
NAME      TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                  AGE     SELECTOR
coredns   ClusterIP   192.168.0.3   <none>        53/UDP,53/TCP,9153/TCP   3h37m   k8s-app=kube-dns

`
But in fact the service is able to reply back reverse query from pods

`
/ # nslookup 192.168.0.1
Server:         192.168.0.3
Address:        192.168.0.3:53

1.0.168.192.in-addr.arpa        name = kubernetes.default.svc.datalake.test

/ # nslookup kubernetes.default.svc.datalake.test
Server:         192.168.0.3
Address:        192.168.0.3:53

*** Can't find kubernetes.default.svc.datalake.test: No answer
` 
Also please note there is also ping that seems to work !! but only if the pod is running on master node - though still no nslookup

`
 ping kubernetes.default
PING kubernetes.default (192.168.0.1): 56 data bytes
64 bytes from 192.168.0.1: seq=0 ttl=64 time=0.131 ms
64 bytes from 192.168.0.1: seq=1 ttl=64 time=0.123 ms
^C
--- kubernetes.default ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 0.123/0.127/0.131 ms
/ # nslookup kubernetes.default
Server:         192.168.0.3
Address:        192.168.0.3:53

** server can't find kubernetes.default: NXDOMAIN

 traceroute -p 53 192.168.0.3
traceroute to 192.168.0.3 (192.168.0.3), 30 hops max, 46 byte packets
 1  coredns.kube-system.svc.datalake.test (192.168.0.3)  0.063 ms  0.007 ms  0.006 ms

`

Here is my env data:

`
 kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.2
Kustomize Version: v4.5.7
Server Version: v1.25.2

`

pod IP range: 192.168.64.0 /18
svc IP range: 192.168.0.0/18

Thanks

**Comment by Mihai-CMM** [comment]
In fact looking at a tcpdump any queries towards svc IP are going to 2 strange IPs but not the IPs of the coredns pods

**Comment by chrisohaver** [comment]
If this _only_ affects the DNS service IP, then it may be that you have nodelocal DNS cache installed, and either it's not managing its iptables rules properly, or nodelocal dns is otherwise unhealthy (nodelocal DNS cache inserts iptables rules that redirect dns queries bound for the coredns service IP).

If this affects _all_ ClusterIP services, it's probably an issue with kube-proxy.

**Comment by Mihai-CMM** [comment]
This happens only with cluster DNS IP only. If i change the svc for coredns and it gets a different IP then all seems fine. If it uses the one that is set as the cluster dns than end game being translated to one of those 2 IPs from the pod rage but who are not in fact the core dns pod IPs . I've looked in iptables-save and i could not find the culprit. I've tried to use IPVS and i hit the same issue. I agree it was most probably something on kube-proxy on my Rocky 9 env. I gave up on kubespray and reinstall a cluster with k0s and all seems ok until now.

Thanks and much appreciate your response
